library/panic_unwind/src/gcc.rs

```diff
@@ -107,7 +107,4 @@ pub unsafe fn cleanup(ptr: *mut u8) -> Box<dyn Any + Send> {
 
 // Rust's exception class identifier.  This is used by personality routines to
 // determine whether the exception was thrown by their own runtime.
-fn rust_exception_class() -> uw::_Unwind_Exception_Class {
-    // M O Z \0  R U S T -- vendor, language
-    0x4d4f5a_00_52555354
-}
+const RUST_EXCEPTION_CLASS: uw::_Unwind_Exception_Class = u64::from_be_bytes(*b"MOZ\0RUST");
```

Comment:
Pre-existing, but should we really reverse the stored order between little-endian and big-endian systems? We currently show `MOZ\0RUST` on big-endian systems and `TSUR\0ZOM` on little-endian systems when interpreted as string/doing a hexdump.

Reply:
I noticed that when I poked at it. :eyes: It is funny to me. All the other runtimes make the same mistake, so I initially decided not to change it. I think I wanted to actually look at the data through a hex dump before doing so.

Reply:
Maybe a relevant datapoint: `windows-drivers-rs`' `wdk-alloc` specifically chose `u32::from_ne_bytes(*b"rust")` for their allocator tag: https://github.com/microsoft/windows-drivers-rs/blob/c3b7c4aa06e43e928f27eb704f76932688802921/crates/wdk-alloc/src/lib.rs#L50

Reply:
Maybe it's not a mistake, maybe the type of this should be thought of as `[u8; 8]` rather than `u64`.

Reply:
The current impl is a mistake if it should be interpreted as `[u8; 8]` as currently little endian systems put the string in reverse order, while big endian systems put it in the correct order.

Reply:
Oh right, that would have to use `from_ne_bytes`... then transmuting back to a `u8` array would always give the same results.

Reply:
i assume it was just someone filling out the u64 literal in left to right order, as the previous comment described.

Reply:
cool, I checked the hexdump emitted and yeah, it is indeed backwards. silly Rust.